From 3f6c78ed9d6ff409d0c738ec1e211e038454a0e7 Mon Sep 17 00:00:00 2001
From: IstvanD <istvan.dezsi@liferay.com>
Date: Wed, 19 May 2021 17:43:17 +0200
Subject: [PATCH] LPS-131699 Add null check

---
 core/event.js | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/core/event.js b/core/event.js
index 6dc8448f62..23c4919d15 100644
--- a/core/event.js
+++ b/core/event.js
@@ -172,7 +172,9 @@
 					}
 
 					function removeListener() {
-						me.removeListener( eventName, listenerFunction );
+						if (me) {
+							me.removeListener( eventName, listenerFunction );
+						}
 					}
 
 					var event = getEntry.call( this, eventName );
