From 213029994cdb8f07eb8299d94a94aeab314c958b Mon Sep 17 00:00:00 2001
From: Norbert Nemeth <norbert.nemeth@liferay.com>
Date: Thu, 11 Mar 2021 16:09:52 +0100
Subject: [PATCH] LPS-127732 remove css classes as well when changing style

---
 plugins/removeformat/plugin.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plugins/removeformat/plugin.js b/plugins/removeformat/plugin.js
index e1e41bb3e..8aa04768a 100644
--- a/plugins/removeformat/plugin.js
+++ b/plugins/removeformat/plugin.js
@@ -35,7 +35,7 @@ CKEDITOR.plugins.removeformat = {
 					range;
 
 				while ( ( range = iterator.getNextRange() ) ) {
-					range.enlarge( CKEDITOR.ENLARGE_INLINE );
+					range.enlarge( CKEDITOR.ENLARGE_ELEMENT );
 
 					// Bookmark the range so we can re-select it after processing.
 					var bookmark = range.createBookmark(),
