From 158cff5bd0b6e6887ee4443887dd00b628d75c88 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marko=20=C4=8Ciko=C5=A1?= <marko.cikos@liferay.com>
Date: Mon, 16 Aug 2021 18:36:20 +0200
Subject: [PATCH] LPS-137425 Don't check selection on focus

---
 core/selection.js | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/core/selection.js b/core/selection.js
index 854796a..65b221b 100644
--- a/core/selection.js
+++ b/core/selection.js
@@ -977,15 +977,6 @@
 				editable.attachListener( editable, 'keydown', disableSelectionChangeForNonEditables, editor );
 			}
 
-			// Always fire the selection change on focus gain.
-			// On Webkit&Gecko (#1113) do this on focusin, because the selection is unlocked on it too and
-			// we need synchronization between those listeners to not lost cached editor._.previousActive property
-			// (which is updated on selectionCheck).
-			editable.attachListener( editable, CKEDITOR.env.webkit || CKEDITOR.env.gecko ? 'focusin' : 'focus', function() {
-				editor.forceNextSelectionCheck();
-				editor.selectionChange( 1 );
-			} );
-
 			// https://dev.ckeditor.com/ticket/9699: On Webkit&Gecko in inline editor we have to check selection when it was changed
 			// by dragging and releasing mouse button outside editable. Dragging (mousedown)
 			// has to be initialized in editable, but for mouseup we listen on document element.
