From 1d81575e3ab3a115d99d932ec65d748dbed35137 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marko=20=C4=8Ciko=C5=A1?= <marko.cikos@liferay.com>
Date: Fri, 21 Oct 2022 15:28:58 +0200
Subject: [PATCH] LPS-166086 Make dialog close button accessible with keyboard

---
 plugins/dialog/plugin.js | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/plugins/dialog/plugin.js b/plugins/dialog/plugin.js
index 5101c888ff..6a38bef707 100644
--- a/plugins/dialog/plugin.js
+++ b/plugins/dialog/plugin.js
@@ -160,7 +160,7 @@ CKEDITOR.DIALOG_STATE_BUSY = 2;
 				'<tr><td role="presentation">' +
 				'<div class="cke_dialog_body" role="presentation">' +
 					'<div id="cke_dialog_title_{id}" class="cke_dialog_title" role="presentation"></div>' +
-					'<a id="cke_dialog_close_button_{id}" class="cke_dialog_close_button" href="javascript:void(0)" title="{closeTitle}" role="button"><span class="cke_label">X</span></a>' +
+					'<a id="cke_dialog_close_button_{id}" class="cke_dialog_close_button close" href="javascript:void(0)" title="{closeTitle}" role="button"><span class="cke_label">X</span></a>' +
 					'<div id="cke_dialog_tabs_{id}" class="cke_dialog_tabs" role="tablist"></div>' +
 					'<table class="cke_dialog_contents" role="presentation">' +
 					'<tr>' +
@@ -425,9 +425,13 @@ CKEDITOR.DIALOG_STATE_BUSY = 2;
 		this.parts.close.on( 'click', function( evt ) {
 			if ( this.fire( 'cancel', { hide: true } ).hide !== false )
 				this.hide();
-			evt.data.preventDefault();
+			evt.data && evt.data.preventDefault();
 		}, this );
 
+		this.on( 'load', function () {
+			this.addFocusable(this.parts.close);
+		});
+
 		// Sort focus list according to tab order definitions.
 		function setupFocus() {
 			var focusList = me._.focusList;
